database/sqlite/sqlite_aclk_alert.c

```diff
@@ -1124,3 +989,140 @@ int get_proto_alert_status(RRDHOST *host, struct proto_alert_status *proto_alert
+        }
+    else
+        return strcmp(active_alerts_a->name, active_alerts_b->name);
+}
+
+void aclk_push_alarm_checkpoint(RRDHOST *host __maybe_unused)
+{
+#ifdef ENABLE_ACLK
+    struct aclk_sync_host_config *wc = host->aclk_sync_host_config;
+    if (unlikely(!wc)) {
+        log_access("ACLK REQ [%s (N/A)]: ALERTS CHECKPOINT REQUEST RECEIVED FOR INVALID NODE", rrdhost_hostname(host));
+        return;
+    }
+
+    //TODO: make sure all pending events are sent.
+    if (rrdhost_flag_check(host, RRDHOST_FLAG_ACLK_STREAM_ALERTS)) {
+        //postpone checkpoint send
+        wc->alert_checkpoint_req++;
+        log_access("ACLK REQ [%s (N/A)]: ALERTS CHECKPOINT POSTPONED", rrdhost_hostname(host));
+        return;
+    }
+
+    //TODO: lock rc here, or make sure it's called when health decides
+    //count them
+    RRDCALC *rc;
+    uint32_t cnt = 0;
+    size_t len = 0;
+    active_alerts_t *active_alerts = NULL;
+
+    foreach_rrdcalc_in_rrdhost_read(host, rc) {
+        if(unlikely(!rc->rrdset || !rc->rrdset->last_collected_time.tv_sec))
+            continue;
+
+        if (rc->status == RRDCALC_STATUS_WARNING ||
+            rc->status == RRDCALC_STATUS_CRITICAL) {
+
+            cnt++;
+        }
+    }
+    foreach_rrdcalc_in_rrdhost_done(rc);
+
+    if (cnt) {
+        active_alerts = callocz(cnt, sizeof(active_alerts_t));
+        cnt = 0;
+        foreach_rrdcalc_in_rrdhost_read(host, rc) {
+            if(unlikely(!rc->rrdset || !rc->rrdset->last_collected_time.tv_sec))
+                continue;
+
+            if (rc->status == RRDCALC_STATUS_WARNING ||
+                rc->status == RRDCALC_STATUS_CRITICAL) {
+
+                active_alerts[cnt].name = (char *)rrdcalc_name(rc);
+                len += string_strlen(rc->name);
+                active_alerts[cnt].chart = (char *)rrdcalc_chart_name(rc);
+                len += string_strlen(rc->chart);
+                active_alerts[cnt].status = rc->status;
+                len++;
+                cnt++;
+            }
+        }
+        foreach_rrdcalc_in_rrdhost_done(rc);
+    }
+
+    BUFFER *alarms_to_hash;
+    if (cnt) {
+        qsort (active_alerts, cnt, sizeof(active_alerts_t), compare_active_alerts);
+
+        alarms_to_hash = buffer_create(len, NULL);
+        for (uint32_t i=0;i<cnt;i++) {
+            buffer_strcat(alarms_to_hash, active_alerts[i].name);
+            buffer_strcat(alarms_to_hash, active_alerts[i].chart);
+            if (active_alerts[i].status == RRDCALC_STATUS_WARNING)
+                buffer_strcat(alarms_to_hash, "W");
+            else if (active_alerts[i].status == RRDCALC_STATUS_CRITICAL)
+                buffer_strcat(alarms_to_hash, "C");
+        }
+    } else {
+        alarms_to_hash = buffer_create(1, NULL);
+        buffer_strcat(alarms_to_hash, "");
+        len = 0;
+    }
+
+    unsigned char hash[SHA256_DIGEST_LENGTH + 1];
+    SHA256_CTX sha256;
+    SHA256_Init(&sha256);
```

Comment:
Hello @MrZammler ,

On newer OpenSSL versions we have warnings like these:

```sh
database/sqlite/sqlite_aclk_alert.c:1112:5: warning: 'SHA256_Init' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 1112 |     SHA256_Init(&sha256);
      |     ^~~~~~~~~~~
In file included from /usr/include/openssl/x509.h:41,
                 from /usr/include/openssl/ssl.h:31,
                 from ./libnetdata/socket/security.h:27,
                 from ./libnetdata/libnetdata.h:651,
                 from ./daemon/common.h:6,
                 from database/sqlite/sqlite_functions.h:6,
                 from database/sqlite/sqlite_aclk_alert.c:3:
/usr/include/openssl/sha.h:73:27: note: declared here
   73 | OSSL_DEPRECATEDIN_3_0 int SHA256_Init(SHA256_CTX *c);
      |                           ^~~~~~~~~~~
database/sqlite/sqlite_aclk_alert.c:1113:5: warning: 'SHA256_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 1113 |     SHA256_Update(&sha256, buffer_tostring(alarms_to_hash), len);
      |     ^~~~~~~~~~~~~
/usr/include/openssl/sha.h:74:27: note: declared here
   74 | OSSL_DEPRECATEDIN_3_0 int SHA256_Update(SHA256_CTX *c,
      |                           ^~~~~~~~~~~~~
database/sqlite/sqlite_aclk_alert.c:1114:5: warning: 'SHA256_Final' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 1114 |     SHA256_Final(hash, &sha256);
      |     ^~~~~~~~~~~~
/usr/include/openssl/sha.h:76:27: note: declared here
   76 | OSSL_DEPRECATEDIN_3_0 int SHA256_Final(unsigned char *md, SHA256_CTX *c);
      |                           ^~~~~~~~~~~~
```

please, take a look when you have a time.

Reply:
Except for these warnings that we need to address, the PR is working as expected.

Reply:
Thank you Thiago! Yes, dumb did not check on openssl 3... changed!